cmd/kubeadm/app/util/version.go

```diff
@@ -92,13 +93,20 @@ func KubernetesReleaseVersion(version string) (string, error) {
 			if body != "" {
 				return "", err
 			}
-			// Handle air-gapped environments by falling back to the client version.
-			klog.Infof("could not fetch a Kubernetes version from the internet: %v", err)
-			klog.Infof("falling back to the local client version: %s", clientVersion)
-			return KubernetesReleaseVersion(clientVersion)
+			if clientVersionErr == nil {
+				// Handle air-gapped environments by falling back to the client version.
+				klog.Infof("could not fetch a Kubernetes version from the internet: %v", err)
```

Comment:
Technically speaking, this one should be warning.

Reply:
Agree. Made them warnings now.